python/perf-kernels/model_benchmarking.py

```diff
@@ -0,0 +1,99 @@
+import json
+import os
+
+
+def load_model_config(config_file='model_configs.json'):
+    """Load all model configurations from a JSON file."""
+    with open(config_file, 'r') as f:
+        return json.load(f)
+
+
+def infer_mnk(model_name, batch_size, seq_len, config_file='model_configs.json'):
+    """Infer M, N, and K dimensions for a given model, batch size, and sequence length."""
+    configs = load_model_config(config_file)
+    if model_name not in configs:
+        raise ValueError(f"Model '{model_name}' not found in {config_file}")
+
+    config = configs[model_name]
+    head_count = config["head_count"]
+    head_dimension = config["head_dimension"]
+
+    # Infer M, N, K based on the feedforward network (FFN) dimensions
+    M = batch_size * seq_len  # Total tokens in a batch
+    K = head_dimension * head_count  # Hidden size (d)
+    N = 4 * K  # FFN dimension is typically 4× hidden size
```

Comment:
Hmm...I think this depends. On llama3-8b, the intermediate size (which is *half* the first FF output) is 14336. So the N dim would be 14336 x 2 or more generally 14336 x intermediate_size.

the intermediate size is a config parameter so we should read it from the json.